### PR TITLE
Add option for trajectory position averaging

### LINKED ISSUE #440: Add trajectory smoothing when importing via MDAnalysis
Support potentially trajectory smoothing when importing an MD trajectory via MDAnalysis. (https://docs.mdanalysis.org/stable/documentation_pages/transformations/positionaveraging.html)

Potential for in theory to add enable / disable support for a variety of transformations. Some of these will be best implemented through Geometry Nodes node groups, but when streaming the trajectory from disk, transformations like averaging will be best done via MDAnalysis as Geometry Nodes won't have access to any frames other than the one that is currently loaded.

### PR DESCRIPTION
**Pull request by BradyAJohnston**
Adds option for trajectory averaging, fixes #440. 

Increasing the `Average` number, accesses that number of frames before and after the current frame, then gets the average position for all of the frames and uses that to update the object's vertex positions.

https://github.com/user-attachments/assets/9fdb6760-5fb1-4ba4-a966-baff22bdb673

### CODE REVIEW

**Comment by BradyAJohnston** [comment]
@yuxuanzhuang if you wanted to have a look and give any feedback before I merge tomorrow

File: molecularnodes/utils.py [hunk 1]
```diff
@@ -40,6 +45,37 @@ def correct_periodic_positions(
...
+
+    frame_a = frame
+
+    if subframes > 0:
+        frame_a = int(frame / (subframes + 1))
+
+    return frame_a
+
+
+def frames_to_average(frame: int, average: int = 0) -> list[int]:
```
Comment (yuxuanzhuang) [inline molecularnodes/utils.py:74]: When `frame=0`, this will also generate negative values; maybe better just filtering those values out?

**Review by yuxuanzhuang — COMMENTED** [review]
It looks great overall! One potential concern could be the performance impact of reading multiple frames (both in average and subframe). It might be worth considering adding a caching mechanism in the next major release to address this.

File: molecularnodes/utils.py (hunk 1 above)
Comment (BradyAJohnston) [reply molecularnodes/utils.py]: good idea, added it in

**Comment by BradyAJohnston** [comment]
I've now added in a caching mechanism. By default it will cache up to 2 frames, so when interpolating without averaging the switch the next frame will only have to fetch the next-next frame. 

When averaging, there will be a window of frames that are collected and cached, so moving forward by 1 frame with a window of 3 frames, will drop the no longer needed frame and fetch only 1 extra frame of positions before averaging. 

The current setup just uses a dictionary under `self.cache` which has the frame number (corresponding to the universe's trajectory frame number) and the numpy array with atom positions for it.

**Pull request merged**